src/rgw/rgw_op.cc

```diff
@@ -5270,20 +5270,25 @@ void RGWCopyObj::execute(optional_yield y)
 
   encode_delete_at_attr(delete_at, attrs);
 
-  // get src object size (cached in obj_ctx from verify_permission())
-  RGWObjState* astate = nullptr;
-  op_ret = src_object->get_obj_state(this, s->obj_ctx, &astate, s->yield, true);
-  if (op_ret < 0) {
-    return;
-  }
 
-  if (!s->system_request) { // no quota enforcement for system requests
-    // enforce quota against the destination bucket owner
-    op_ret = dest_bucket->check_quota(this, user_quota, bucket_quota,
-				      astate->accounted_size, y);
+  uint64_t obj_size = 0;
+  {
+    // get src object size (cached in obj_ctx from verify_permission())
```

Comment:
curious why the indentation, was this intended to be a `lock_guard()` block?

Reply:
i like that it limits the scope of `RGWObjState* astate` so it can't be referenced after a possible invalidate